### ISSUE TITLE
Tabs vs space indentation: Add option to detect

### ISSUE THREAD

**Issue by Bolpat** [notepad-plus-plus/notepad-plus-plus#13860]
### Description of the Issue
Notepad++ has a setting to tell if tabs should be replaced by spaces (and if so, by how many). This is great for new files, but bad for existing files that already use the opposite of the setting.

It would be great if Notepad++ could detect:
* if a file uses tabs or spaces for indentation — and
* for indentation with spaces, the tabulation width.

It need not be perfect, it just has to guess right in most cases. It’s similar to detection of line endings.

As an example, Visual Studio 2022 has such a detection.

For a suggestion on how to implement this:
* Use the application’s setting unless the file contains lines which have a tab or 2 spaces as their first characters and all those lines start with the same sequence (1 tab or 2 spaces, not mixed). That determines tabs vs spaces. (Maybe exclude lines that consist of only of white space.)
* If the tabulation is detected as spaces, guess the tab width as the number of initial spaces of the first line that starts with 2 or more spaces, then round that number down to 2, 4, or 8.

I’d guess it’s important to rule out a tabulation width of 1 space for false positives like (documentation) comments of the form:
```c
/* next line starts with 1 space, but it’s not indented by any notion!
 */
```

**Comment by alankilborn** [comment]
Check out the [EditorConfig](https://editorconfig.org/) plugin.

**Comment by daxliniere** [comment]
Yes, I love N++, but I have been having real headaches lately when copy/pasting YAML-format data from N++ into the config page on a virtual machine. It keeps converting all spaces to tabs, which is a character not supported in YAML.

**Comment by alankilborn** [comment]
> It keeps converting all spaces to tabs

@daxliniere Please explain in more detail; there should be no conversion on paste.

**Comment by daxliniere** [comment]
Hi @alankilborn, thanks for your reply. Let me dig into this further, it's possible I have just caught a mistake in my method.

**Comment by Bolpat** [comment]
> Check out the [EditorConfig](https://editorconfig.org/) plugin.

How would that plugin help be with making N++ detect indentation?

**Comment by Bolpat** [comment]
> Check out the [EditorConfig](https://editorconfig.org/) plugin.

Complete non-starter. This is a basic feature that shouldn’t require a plugin and configuration for every folder. I have files in various folders and I don’t want to add `.editorconfig`s all over my disk to set tabs vs spaces for single files. Just auto-detect them.

**Comment by Bolpat** [comment]
Alt least, when adding a new line (pressing the Enter key), use the indentation as-is of the previous line. If it’s spaces, copy spaces. If it’s tabs, copy the tabs. If my file uses tabs, why would I want space-indentation in _one_ line?
